AIDA-665

jakevb opened the pull request:
Check README for clarity and accuracy explaining the different types of ontology.  Also updated latest versions of SeedlingOntology, SeedlingOwlOntology, and LDCOwlOntology.

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+# AIF Ontologies
```
```suggestion
# AIDA Ontologies
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+## AidaDomainOntologiesCommon
+
+A collection of top level classes most other ontologies reference as the highest level class.  Additionally, there are
```
```suggestion
A collection of top level classes all Domain Ontologies reference.  Additionally, there are
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+## AidaDomainOntologiesCommon
+
+A collection of top level classes most other ontologies reference as the highest level class.  Additionally, there are
+also a few universal classes that include important properties such as the CanHaveNumericValue class.  This ontology was
```
```suggestion
also a few universal classes that include important properties such as the CanHaveNumericValue class.  This ontology is a hand-written source file and is not generated.
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+A collection of top level classes most other ontologies reference as the highest level class.  Additionally, there are
+also a few universal classes that include important properties such as the CanHaveNumericValue class.  This ontology was
+created by individually selecting those class values and not from any other source of data.
```
```suggestion

```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+## EntityOntology
+
+A collection of Entity classes determined by the Ontology working group.  Entity referring to
```
```suggestion
A collection of Entity classes determined by the AIDA Ontology Working Group. The term 'Entity' refers to
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+## EntityOntology
+
+A collection of Entity classes determined by the Ontology working group.  Entity referring to
+a physical being or intangible concept that either interacts or is subject to interaction.  This in combination with Events
```
```suggestion
a physical being or intangible concept that either interacts or is subject to interaction.  This, in combination with Events
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+A collection of Entity classes determined by the Ontology working group.  Entity referring to
+a physical being or intangible concept that either interacts or is subject to interaction.  This in combination with Events
+and Relations ontologies contain the vocabulary that will be used for defining messages post-M18 evaluation.
```
```suggestion
and Relations ontologies, contains the vocabulary that will be used for defining messages post-M18 evaluation.
```

craigwarsaw replied:
With the new organization that I recommended in the top-level comment for this PR, you can remove the starting with sentence "This in combination with" and replace it with a similar sentence in the main "Program-Wide Ontology Files" section. Like "The Entities, Events, and Relations ontologies contain..."

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+## EventOntology
+
+A collection of Event classes determined by the Ontology working group.  Event referring to
+an experience that occurs when two or more Events interact in some way.  Some examples being building weaponry, signing a
```
```suggestion
something that occurs when two or more Entities interact in some way.  Some examples being building weaponry, signing a
```

craigwarsaw replied:
I think you mean "Entities interact in some way", not "Events interact".

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+A collection of Event classes determined by the Ontology working group.  Event referring to
+an experience that occurs when two or more Events interact in some way.  Some examples being building weaponry, signing a
+treaty, and the moving of people from one place to another. This in combination with Entity
```
```suggestion
treaty, and the moving of people from one place to another. This, in combination with Entity
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+A collection of Event classes determined by the Ontology working group.  Event referring to
+an experience that occurs when two or more Events interact in some way.  Some examples being building weaponry, signing a
+treaty, and the moving of people from one place to another. This in combination with Entity
+and Relations ontologies contain the vocabulary that will be used for defining messages post-M18 evaluation.
```
```suggestion
and Relations ontologies, contains the vocabulary that will be used for defining messages post-M18 evaluation.
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+## InterchangeOntology
+
+The main purpose of this ontology is to contain the Interchange vocabulary and to be used by performers using strict OWL
```
```suggestion
This ontology contains the Interchange vocabulary and is written in strict OWL. 
```

craigwarsaw replied:
I think "The main purpose of this ontology is to contain the Interchange vocabulary" is correct, but I'm not sure that the rest is correct.  Let's discuss to help me understand this statement, and the following sentence starting with "In particular".

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+## InterchangeOntology
+
+The main purpose of this ontology is to contain the Interchange vocabulary and to be used by performers using strict OWL
+as reference to the classes and properties that are used in our not-strict OWL ontologies. In particular, it contains
```
```suggestion
In particular, it contains
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+The main purpose of this ontology is to contain the Interchange vocabulary and to be used by performers using strict OWL
+as reference to the classes and properties that are used in our not-strict OWL ontologies. In particular, it contains
+the OWL definitions for domain and range which are usually omitted from the not-strict OWL ontologies.
```
```suggestion
the OWL definitions for domain and range which describe how each property interacts with defined classes.
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+# LDCOntology
+
+The ontology was given by LDC as an evolution of the SeedlingOntology and will be used for the M18 evaluation.  This will eventually
```
```suggestion
This Domain Ontology was given by LDC as an evolution of the SeedlingOntology and will be used for the M18 evaluation. 
```

craigwarsaw replied:
In the new structure that I recommended in the top-level comment of this PR, put this sentence in the main section".
Also change "given" to "provided"

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+# LDCOntology
+
+The ontology was given by LDC as an evolution of the SeedlingOntology and will be used for the M18 evaluation.  This will eventually
+replace SeedlingOntology once the original LDC data has been removed from our programs and updated with the newest version.
```
```suggestion

```

craigwarsaw replied:
"once the original seedling data"

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+## LDCOwlOntology
+
+The ontology was given by LDC as an evolution of the SeedlingOwlOntology and will be used for the M18 evaluation.  This will eventually
```
```suggestion
This Domain Ontology was given by LDC as an evolution of the SeedlingOwlOntology and will be used for the M18 evaluation.
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+## LDCOwlOntology
+
+The ontology was given by LDC as an evolution of the SeedlingOwlOntology and will be used for the M18 evaluation.  This will eventually
+replace SeedlingOwlOntology once the original LDC data has been removed from our programs and updated with the newest version.
```
```suggestion

```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+## EventOntology
+
+A collection of Event classes determined by the Ontology working group.  Event referring to
```
```suggestion
A collection of Event classes determined by the Ontology working group.  The term 'Event' refers to
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+## RelationOntology
+
+A collection of Relation Type classes determined by the Ontology working group.  Relations referring to
```
```suggestion
A collection of Relation Type classes determined by the Ontology Working Group. The term 'Relation' refers to
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+## RelationOntology
+
+A collection of Relation Type classes determined by the Ontology working group.  Relations referring to
+A specification of the connection between two or more Entities. This includes such things as the material used to create
```
```suggestion
a connection between two or more Entities. This includes such things as the material used to create
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+A collection of Relation Type classes determined by the Ontology working group.  Relations referring to
+A specification of the connection between two or more Entities. This includes such things as the material used to create
+a weapon, the nationality of an individual and the title an employee holds. This in combination with Entities
```
```suggestion
a weapon, the nationality of an individual and the title an employee holds. This, in combination with Entities
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+A collection of Relation Type classes determined by the Ontology working group.  Relations referring to
+A specification of the connection between two or more Entities. This includes such things as the material used to create
+a weapon, the nationality of an individual and the title an employee holds. This in combination with Entities
+and Events ontologies contain the vocabulary that will be used for defining messages post-M18 evaluation.
```
```suggestion
and Events ontologies, contains the vocabulary that will be used for defining messages post-M18 evaluation.
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+## SeedlingOntology
+
+The ontology created from the first cut of data from LDC and was primarily used by performers during the M9 evaluation.
```
```suggestion
This Domain Ontology was created from the first cut of data from LDC and was primarily used by performers during the M9 evaluation.
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,66 @@
+
+## SeedlingOwlOntology
+
+The ontology created from the first cut of data from LDC and was primarily used by performers during the M9 evaluation.
```
```suggestion
This Domain Ontology was created from the first cut of data from LDC and was primarily used by performers during the M9 evaluation.
```

jakevb commented:
Craig, I have incorporated almost all of the changes both you and Eddie have suggested.  The only one that might still need some clarification is your question regarding the InterchangeOntology.  I have updated that section to improve clarity and incorporate Eddie's suggestions, but if you still have questions, we can discuss what might need to be changed.

craigwarsaw commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,62 @@
+# AIDA Ontologies
+
+The following is a listing of the ontologies found at AIDA-Interchange-Format/src/main/resources/com/ncc/aif/ontologies.
+Each ontology will contain a short description of what they are built from, what they are used for,
```
I think you meant "Each section in this README contains..."

craigwarsaw commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,62 @@
+
+# Program-Wide Ontology Files
+
+The Entities, Events, and Relations ontologies contain the vocabulary that will be used for definning messages post-M18 evaluation.
```
MIsspelled "defining"

craigwarsaw commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,62 @@
+## RelationOntology
+
+A collection of Relation Type classes determined by the AIDA Ontology Working Group. The term 'Relation' refers to
+A connection between two or more Entities. This includes such things as the material used to create
```
The first letter of this line "A" should be "a"

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,62 @@
+## Differences Between *Ontology and *OwlOntology Files
+
+Some ontologies have a corresponding file with a similar name such as SeedlingOntology and SeedlingOwlOntology. In these
+cases, the *OwlOntology files signifies that the rdf file is compatible with strict OWL, in particular using the correct syntax
```
```suggestion
cases, the *OwlOntology file uses strict OWL and won't include third-party properties/classes 
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,62 @@
+
+Some ontologies have a corresponding file with a similar name such as SeedlingOntology and SeedlingOwlOntology. In these
+cases, the *OwlOntology files signifies that the rdf file is compatible with strict OWL, in particular using the correct syntax
+in lieu of *Ontology file properties like schema:rangeIncludes. The *Ontology files are not restricted and can make use of third-party properties to simplify definitions.
```
```suggestion
like schema:rangeIncludes. The *Ontology files are not restricted and make use of third-party properties/classes to simplify definitions.
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,62 @@
+in lieu of *Ontology file properties like schema:rangeIncludes. The *Ontology files are not restricted and can make use of third-party properties to simplify definitions.
+Both the *OwlOntology and *Ontology files define the same vocabulary.
+
+# Common Ontology Files
```
```suggestion
# AIF Ontology Files
```

ecurley-nextcentury commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,62 @@
+
+# Seedling Ontology Files
+
+This Domain Ontology created from the first cut of data from LDC and was primarily used by performers during the M9 evaluation.
```
```suggestion
This Domain Ontology was created from the first cut of data from LDC and was primarily used by performers during the M9 evaluation.
```

craigwarsaw commented on src/main/resources/com/ncc/aif/ontologies/README.md:
```diff
@@ -0,0 +1,64 @@
+
+### InterchangeOntology
+
+This ontology contains the Interchange vocabulary and is written in strict OWL. In particular, it contains
```
After discussion with @jakevb, I think I understand where I was confused.  I recommend changing the entire sentence starting from "In particular..." to the following.

It defines the structure of the vocabulary, including the hierarchy of terms (via the "domain" schema definition) and the possible datatypes of terms (via the "range" schema definition).

@ecurley-nextcentury - do you agree with the rewording of this?

ecurley-nextcentury replied:
All RDF is represented as triples (**subject**, **predicate**, **object**). "Domain" refers to what types a particular **predicate** can have as **subject** and "range" refers to what types that **predicate** can have as **object**. If these terms are confusing things, we should just remove them. We can probably change the description to something like, "This ontology defines the Interchange vocabulary and how the defined classes and properties may interact". I'll leave it to you and Jake to come up with something that works. Let me know if you have any questions.

craigwarsaw replied:
Your recommendation is good with me.

jakevb replied:
Sounds good.  I incorporated Eddie's suggested change.
